### DISCUSSION TITLE
No mono AAC+

### DISCUSSION THREAD

**Discussion by KevanGP** [savonet/liquidsoap#2442] [User Support]
**Describe the bug**
A clear and concise description of what the bug is. Add some script and log extract so we can see what is happening. Don't forget to add some context although not necessarily your whole script..
Hi!
Sorry if this is already reported. But I'm trying to get mono AAC+ audio out of a stream?
Here is my AAC+ code from the script for streaming to an Icecast server. It works with stereo, so that's what I'm pasting below.
"%fdkaac(channels=2, samplerate=44100,
bandwidth="auto", bitrate=64,
afterburner=true, aot="mpeg4_he_aac",
transmux="adts", sbr_mode=false)"
Changing channels to 1 instead of 2 produces an error and Liquidsoap fails to start. Is there another setting which must be changed? I'm using aot="mpeg4_he_aac", not aot="mpeg4_he_aac_v2", because V2 ads weired parametric stereo effects to the sound. It would be nice if AAC+ was easier to configure. MP3 variable bitrate I couldn't get to work with mono audio either. Mono sounds better at lower bit rates for MP3, and AAC+ sometimes as well.
**To Reproduce**
A minimal script to reproduce the issue is the most efficient way to have your bug addressed and fixed very quickly!
"%fdkaac(channels=1, samplerate=44100, bandwidth="auto", bitrate=64, afterburner=false, aot="mpeg4_he_aac", transmux="adts", sbr_mode=false)"
**Expected behavior**
It should start up properly and broadcast the AAC+ stream in mono.
**Version details**
 - OS: [e.g. Debian, OSX]
 - Version [e.g. `1.3.4`]
Latest stable Liquidsoap, Linux 22.04
**Install method**
Did you install via `opam`? Using your distribution's packages? From source?
Opam

**Comment by toots** [comment]
This script works here:
```ruby
enc = %fdkaac(
  afterburner=false,
  aot="mpeg4_he_aac",
  bitrate=64,
  channels=1,
  samplerate=44100,
  sbr_mode=false,
  transmux="adts"
)

output.file(enc, "/tmp/bla.aac", sine())
```
